index.js

```diff
@@ -17,70 +19,111 @@ const productB = {
     commission: 75
 }
 
-// Light and dark mode switch
-const toggleBtn = document.querySelector('.toggle-mode')
-toggleBtn.addEventListener('click', function () {
-    document.body.classList.toggle('light')
-    toggleBtn.classList.toggle('fa-moon')
-    toggleBtn.classList.toggle('fa-sun')
-})
-
-// Setup
-// The sales buttons
-const paBtn = document.getElementById('btn-productA')
-const pbBtn = document.getElementById('btn-productB')
+// *** We have these emojis hardcoded in the html, so I don't believe these assignments are necessary - Matt
 // Show the correct emoji on those sales buttons
-paBtn.textContent = productA.emoji
-pbBtn.textContent = productB.emoji
-// Let them listen for clicks & when they're hit fire a sales function
-paBtn.addEventListener('click', function () { fixSale(productA) })
-pbBtn.addEventListener('click', function () { fixSale(productB) })
+// paBtn.textContent = productA.emoji
+// pbBtn.textContent = productB.emoji
 
-// Showing the sales results in the HTML
-const salesHeader = document.getElementById('sales-header')
-const salesBar = document.getElementById('sales-bar')
-const achievementsHeader = document.getElementById('achievements-header')
-const achievementsBar = document.getElementById('achievements-bar')
-// initally they're empty (need to refactor this when we're using local storage)
-salesBar.textContent = achievementsBar.textContent = ''
+// Fetch sales data from local storage (if available)
+function fetchSalesData() {
+    if (localStorage.getItem('salesData')) { // if there is saved sales data available 
+        return JSON.parse(localStorage.getItem('salesData')); // return the saved sales data
+    } else { // otherwise
+       return { // return an object with these starting values:
+            totalSales: '',
+            timesClicked: 0,
+            achievementsUnlocked: 0,
+            achievementsBadge: '',
+            achievementsBadgeTwo: false,
+            totalRevenue: 0,
+            totalCommission: 0,
```

Comment:
I agree, would it be better to create a file for the data and hold it in an object and then that way we could import the object and use it for reset purposes?

Reply:
Actually, replacing that code with the resetSalesData function causes an error. The else clause in the fetchSalesData is where a new salesData object is returned. This code runs if there is no salesData object available in local storage (so, first time running the app in a browser). The resetSalesData is actually resetting the salesData objects to the original values, so it needs that object to be available to work properly. I know, it's confusing. I played around with that functionality for quite a while before settling on this particular solution, but I think I'll take another crack at it and see if it can be streamlined better. Maybe I ought to rename that reset func to "initializeSalesData" instead. That might reflect better what it's actually doing.